modules/seastate/src/SeaState_Input.f90

```diff
@@ -589,129 +580,56 @@ subroutine SeaStateInput_ProcessInitData( InitInp, p, InputFileData, ErrStat, Er
-         call SetErrStat( ErrStat2, ErrMsg2, ErrStat, ErrMsg,RoutineName)
-         if ( ErrStat >= AbortErrLev ) return
-
-      WaveModIn               = InputFileData%Waves%WaveMod
-
-   end if ! LEN_TRIM(InputFileData%Waves%WaveModChr)
-
-
-   if ( WaveModIn < 0 .OR. WaveModIn > 7 ) then
+   SELECT CASE(InputFileData%WaveMod)
+      CASE(WaveMod_None)
+      CASE(WaveMod_Regular)
+      CASE(WaveMod_RegularUsrPh)
+      CASE(WaveMod_JONSWAP)
+      CASE(WaveMod_WhiteNoise)
+      CASE(WaveMod_UserSpctrm)
+      CASE(WaveMod_ExtElev)
+      CASE(WaveMod_ExtFull)
+      CASE(WaveMod_UserFreq)
+      CASE DEFAULT
          call SetErrStat( ErrID_Fatal,'WaveMod must be 0, 1, 1P#, 2, 3, 4, 5, 6, or 7',ErrStat,ErrMsg,RoutineName)
          return
-   end if
+   END SELECT
 
-      ! Linearization Checks
-   ! LIN-TODO:
-   !errors if:
-   !if (                                                                   &
-   !     (WaveModIn /= 0)                                             .or. &
-   !     (InputFileData%Waves2%WvDiffQTFF /= .false.)                       .or. &
-   !     (InputFileData%Waves2%WvSumQTFF /= .false.)                        .or. &
-   !     (InputFileData%PotMod /= 0 .or. InputFileData%PotMod /=1)                .or. &
-   !     (InputFileData%WAMIT%ExctnMod /=0 .or. InputFileData%WAMIT%ExctnMod /=2) .or. &
-   !     (InputFileData%WAMIT%RdtnMod  /=0 .or. InputFileData%WAMIT%RdtnMod  /=2) .or. &
-   !     (InputFileData%WAMIT2%MnDrift /=0)                                 .or. &
-   !     (InputFileData%WAMIT2%NewmanApp /= 0)                              .or. &
-   !     (InputFileData%WAMIT2%SumQTF /= 0 )                                     ) then
-   !
-   !end if
 
 
    ! WaveStMod - Model switch for stretching incident wave kinematics to instantaneous free surface.
 
    ! TODO: We are only implementing WaveStMod = 0 (No stretching) at this point in time. 1 Mar 2013 GJH
    ! All three methods of wave stretching tentatively implemented.
 
-   IF ( InputFileData%Waves%WaveMod /= 0 .AND. InputFileData%Waves%WaveMod /= 6 ) THEN
-      IF ( (InputFileData%Waves%WaveStMod /= 0) .AND. (InputFileData%Waves%WaveStMod /= 1) .AND. &
-           (InputFileData%Waves%WaveStMod /= 2) .AND. (InputFileData%Waves%WaveStMod /= 3) ) THEN
+   IF ( InputFileData%WaveMod /= WaveMod_None .AND. InputFileData%WaveMod /= WaveMod_ExtFull ) THEN
+      IF ( (InputFileData%WaveStMod /= 0) .AND. (InputFileData%WaveStMod /= 1) .AND. &
+           (InputFileData%WaveStMod /= 2) .AND. (InputFileData%WaveStMod /= 3) ) THEN
          CALL SetErrStat( ErrID_Fatal,'WaveStMod must be 0, 1, 2, or 3.',ErrStat,ErrMsg,RoutineName)
          RETURN
       END IF
-   ELSE ! Wave stretching is not supported when WaveMod = 0 or 6.
-      InputFileData%Waves%WaveStMod = 0_IntKi
+   ELSE ! Wave stretching is not supported when WaveMod = 0 (WaveMod_None) or 6 (WaveMod_ExtFull).
+      InputFileData%WaveStMod = 0_IntKi
```

Comment:
This is not consistent with the check in HydroDyn, where it says WaveStMod can be 0, 1, or 3 for WaveMod_ExtFull.

Reply:
@bjonkman This check in SeaState should be corrected following the check in HydroDyn above. WaveStMod=0, 1, and 3 should be allowed with WaveMod_ExtFull.

Reply:
Addressed in 67c77c: updated with the check from HydroDyn